types/shared.d.ts

```diff
@@ -374,31 +376,18 @@ export interface IStripeConnectInitParams {
   locale?: string;
 }
 
-type ConnectHTMLElementRecordFallback = {
-  [key in string]: HTMLElement | null;
-};
-type ConnectHTMLElementRecordBase = {
-  [tagName in ConnectElementTagName]: HTMLElement;
-};
+type ConnectElementCustomMethods = typeof ConnectElementCustomMethodConfig;
 
-interface ConnectHTMLElementRecord
-  extends ConnectHTMLElementRecordBase,
-    ConnectHTMLElementRecordFallback {
-  "payment-details": HTMLElement & {
-    setPayment: (payment: string | undefined) => void;
-    setOnClose: (listener: (() => void) | undefined) => void;
-  };
-  "account-onboarding": HTMLElement & {
-    setFullTermsOfServiceUrl: (termOfServiceUrl: string | undefined) => void;
-    setRecipientTermsOfServiceUrl: (
-      recipientTermsOfServiceUrl: string | undefined
-    ) => void;
-    setPrivacyPolicyUrl: (privacyPolicyUrl: string | undefined) => void;
-    setSkipTermsOfServiceCollection: (
-      skipTermsOfServiceCollection: boolean | undefined
-    ) => void;
-    setOnExit: (listener: (() => void) | undefined) => void;
+type ConnectHTMLElementRecordBase = {
+  [K in keyof ConnectElementCustomMethods]: HTMLElement &
+    ConnectElementCustomMethods[K];
+} &
+  {
+    [tagName in ConnectElementTagName]: HTMLElement;
   };
+
+interface ConnectHTMLElementRecord extends ConnectHTMLElementRecordBase {
+  [key: string]: HTMLElement | null;
```

Comment:
Can we remove this | null, does it cause problems? It would be ideal to avoid those type issues!

<img width="466" alt="image" src="https://github.com/stripe/connect-js/assets/95381655/852f3fe4-0406-40dd-bb94-134e54d32a40">

Reply:
I think I made it `HTMLElement | null` if the tag name isn't one of the connect element tag names since the element may not exist, is it more preferable to not have `null`?

Reply:
yes, so the code I wrote ^ compiles without typescript errors